docs/root/navigating_the_codebase.md

```diff
@@ -0,0 +1,176 @@
+APIs.
+
+The [benchmarks](../../benchmarks) directory contains a test suite built on top
+of Nighthawk's integration tests that allows users to develop their own
+benchmarks. See the [benchmarks documentation](../../benchmarks/README.md) for
+more details.
+
+The [ci](../../ci) directory contains configuration and scripts used when
+executing continuous integration pipelines for Nighthawk.
+
+The [docs](../../docs) contain Nighthawk's documentation.
+
+The [extensions_build_config.bzl](../../extensions_build_config.bzl) selects the
+extensions the Envoy used by Nighthawk is built with. See [Building Envoy with
+Bazel](https://github.com/envoyproxy/envoy/blob/main/bazel/README.md) for
+details. Nighthawk uses Envoy both in the client code to send requests and the
+in the [Nighthawk test server](overview.md#nighthawk_test_server) to respond to
+requests.
+
+The [include](../../include) directory contains C++ header files (declarations)
+of Nighthawk components. These aren't considered to be part of the public API
+and may change as Nighthawk develops. See [the
+include](#the-include-and-source-directories) section for more details about
+this directory.
+
+The [internal_proto](../../internal_proto) directory contains protocol buffer
+messages that are internal to Nighthawk.
+
+The [samples](../../samples) directory contains samples in the
+[fortio](https://github.com/fortio/fortio#report-only-ui) data format. See the
+[README.md](../../README.md#visualizing-the-output-of-a-benchmark) for more
+details.
+
+The [source](../../source) directory contains C++ implementations (definitions)
+of Nighthawk components declared in the **include** directory as well as the
+build targets for the Nighthawk binaries. See [the
+source](#the-include-and-source-directories) section for more details about
+this directory.
+
+The [support](../../support) directory contains tools used when developing
+Nighthawk, e.g. git pre-commit hooks. See its
+[README.md](../../support/README.md) for more details.
+
+The [test](../../test) contains Nighthawk's unit and integration tests. The
+integration tests are located in the [integration](../../test/integration)
+subdirectory.
+
+The [tools](../../tools) directory contains utilities used to upkeep the
+Nighthawk codebase, e.g. tools that enforce file formatting rules, etc.
+
+The [WORKSPACE](../../WORKSPACE) file contains Bazel [workspace
+rules](https://docs.bazel.build/versions/main/be/workspace.html) that define
+dependencies external to this repository.
+
+## The APIs
+
+The **api** directory has the following subdirectories:
+
+```
+.
+└── api
+    ├── adaptive_load
+    ├── client
+    ├── distributor
+    ├── request_source
+    ├── server
+    └── sink
+```
+
+The [adaptive_load](../../api/adaptive_load) directory contains protocol buffer
+messages that are used to configure the [adaptive load
+controller](adaptive_load_controller.md).
+
+The [client](../../api/client) directory contains the main API of the Nighthawk
```

Comment:
suggest including a note about traffic generation or load generation here

Reply:
Sorry, I am straggling to figure out how to incorporate it into the sentence, mostly because I am not sure what we are trying to say. Would you be able to provide an example?

Reply:
The client directory contains the main API for Nighthawk's traffic/load generation - both the CLI and the gRPC service definition...

My point is that nighthawk's codebase actually is doing a few things, so clarifying that the main API of Nighthawk is for traffic/load generation is a helpful note.

Reply:
Thank you for clarifying, done in 507e88b80f1c90d5cc9778c9e825915e147ad592.